src/iosrv/init.c

```diff
@@ -256,7 +256,73 @@ dss_topo_init()
 }
 
 static int
-server_init()
+abt_max_num_xstreams(void)
+{
+	char   *env;
+
+	env = getenv("ABT_MAX_NUM_XSTREAMS");
+	if (env == NULL)
+		env = getenv("ABT_ENV_MAX_NUM_XSTREAMS");
+	if (env != NULL)
+		return atoi(env);
+	return 0;
+}
+
+static int
+set_abt_max_num_xstreams(int n)
+{
+	char   *name = "ABT_MAX_NUM_XSTREAMS";
+	char   *value;
+	int	rc;
+
+	D_ASSERTF(n > 0, "%d\n", n);
+	D_ASPRINTF(value, "%d", n);
+	if (value == NULL)
+		return -DER_NOMEM;
+	D_INFO("Setting %s to %s\n", name, value);
+	rc = setenv(name, value, 1 /* overwrite */);
+	D_FREE(value);
+	if (rc != 0)
+		return daos_errno2der(errno);
+	return 0;
+}
+
+static int
+abt_init(int argc, char *argv[])
+{
+	int	nrequested = abt_max_num_xstreams();
+	int	nrequired = 1 /* primary xstream */ + DSS_XS_NR_TOTAL;
```

Comment:
I think we should use dss_sys_xs_nr instead of "1", although they are the same for now?

Reply:
The primary xstream is the one executing main; all xstreams we created for sys and targets are secondary xstreams. DSS_XS_NR_TOTAL == # of secondary xstreams. Right?